examples/EquivariantFilter.h

```diff
@@ -0,0 +1,351 @@
+    throw std::invalid_argument(
+        "Initial covariance dimensions must match the degrees of freedom");
+  }
+
+  if (n_cal < 0) {
+    throw std::invalid_argument(
+        "Number of calibration states must be non-negative");
+  }
+
+  if (m <= 1) {
+    throw std::invalid_argument(
+        "Number of direction sensors must be at least 2");
+  }
+
+  static_assert(has_lift<Geometry, M>::value,
+                "Geometry must implement static lift(const M&, const Input&)");
+
+  static_assert(has_stateTransitionMatrix<Geometry>::value,
+                "Geometry must define static stateTransitionMatrix(Input, "
+                "double, GType)");
+  static_assert(has_groupAction<Geometry>::value,
+                "Geometry must define groupAction(GType, MType)");
+  static_assert(
+      has_stateMatrixA<Geometry>::value,
+      "Geometry must define static stateMatrixA(const GType&, const Input&)");
+  static_assert(has_inputMatrix<Geometry>::value,
+                "Geometry must define static inputMatrix(GType)");
+  static_assert(has_processNoise<Geometry>::value,
+                "Geometry must define static processNoise(const Input&)");
+  static_assert(has_inputMatrixBt<Geometry>::value,
+                "Geometry must define static inputMatrixBt(GType)");
+  static_assert(
+      has_measurementMatrixC<Geometry>::value,
+      "Geometry must define static measurementMatrixC(const Unit3&, int)");
+  static_assert(has_outputMatrixDt<Geometry>::value,
+                "Geometry must define static outputMatrixDt(int, GType)");
+
+  // Compute differential of phi
+  Dphi0 = stateActionDiff(xi_0);
+  InnovationLift = Dphi0.completeOrthogonalDecomposition().pseudoInverse();
+}
+/**
+ * Computes the internal group state to a physical state estimate
+ * @return Current state estimate
+ */
+template <typename G, typename M, typename Geometry>
+M EqF<G, M, Geometry>::stateEstimate() const {
+  return Geometry::groupAction(X_hat, xi_0);
+}
+/**
+ * Implements the prediction step of the EqF using system dynamics and
+ * covariance propagation and advances the filter state by symmtery-preserving
+ * dynamics.Uses a Lie group integrator scheme for discrete time propagation
+ * @param u Angular velocity measurements
+ * @param dt time steps
+ * Updated internal state and covariance
+ */
+template <typename G, typename M, typename Geometry>
+void EqF<G, M, Geometry>::propagation(const typename Geometry::Input& u,
+                                      double dt) {
+  M state_est = stateEstimate();            // General manifold
+  Vector L = Geometry::lift(state_est, u);  // Lifted vector in tangent space
+
+  Matrix Phi = Geometry::stateTransitionMatrix(u, dt, X_hat);
+  Matrix Bt = Geometry::inputMatrixBt(X_hat);
+
+  Matrix Q = Geometry::processNoise(u);  // Replaces blockDiag(...)
+
+  X_hat = traits<G>::Compose(X_hat, traits<G>::Expmap(L * dt));
+  Sigma = Phi * Sigma * Phi.transpose() + Bt * Q * Bt.transpose() * dt;
```

Comment:
This is already done in the ManifoldEKF, specialized to groups in the LieGroupEKF, which is the main reason to derive form that class.